core/trino-main/src/main/java/io/trino/sql/planner/AnonymizePlanRewriter.java

```diff
@@ -0,0 +1,785 @@
+    {
+        return nodes.stream()
+                .map(context::rewrite)
+                .collect(toImmutableList());
+    }
+
+    private PartitioningScheme anonymize(PartitioningScheme scheme)
+    {
+        Partitioning anonymizedPartitioning = Partitioning.create(
+                scheme.getPartitioning().getHandle(),
+                scheme.getPartitioning().getColumns().stream()
+                        .map(this::anonymize)
+                        .collect(toImmutableList()));
+
+        return new PartitioningScheme(
+                anonymizedPartitioning,
+                anonymize(scheme.getOutputLayout()),
+                scheme.getHashColumn().map(this::anonymize),
+                scheme.isReplicateNullsAndAny(),
+                scheme.getBucketToPartition());
+    }
+
+    private PlanNodeStatsEstimate anonymize(PlanNodeStatsEstimate statsEstimate)
+    {
+        return new PlanNodeStatsEstimate(
+                statsEstimate.getOutputRowCount(),
+                anonymize(statsEstimate.getSymbolStatistics()));
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> Map<Symbol, T> anonymize(Map<Symbol, T> symbols)
+    {
+        ImmutableMap.Builder<Symbol, T> anonymizedSymbols = ImmutableMap.builder();
+        symbols.forEach((key, value) -> {
+            T anonymizedValue = value;
+            if (value instanceof Symbol) {
+                anonymizedValue = (T) anonymize((Symbol) value);
+            }
+            else if (value instanceof Expression) {
+                anonymizedValue = (T) anonymize((Expression) value);
+            }
+            else if (value instanceof Aggregation) {
+                anonymizedValue = (T) anonymize((Aggregation) value);
+            }
+            else if (value instanceof ColumnHandle) {
+                anonymizedValue = (T) new ColumnHandle() {};
+            }
+
+            anonymizedSymbols.put(anonymize(key), anonymizedValue);
+        });
+        return anonymizedSymbols.build();
+    }
+
+    private IndexHandle anonymize(IndexHandle handle)
+    {
+        return new IndexHandle(
+                handle.getCatalogName(),
+                new ConnectorTransactionHandle() {},
+                new ConnectorIndexHandle() {});
+    }
+
+    private TableHandle anonymize(TableHandle handle)
+    {
+        TableSchema tableSchema = metadata.getTableSchema(session, handle);
+        return new TableHandle(
+                handle.getCatalogName(),
+                new AnonymizedTableHandle(
+                        new QualifiedObjectName(
+                                anonymize(tableSchema.getQualifiedName().getCatalogName()),
+                                anonymize(tableSchema.getQualifiedName().getSchemaName()),
+                                anonymize(tableSchema.getQualifiedName().getObjectName())),
+                        handle.getConnectorHandle().getClass().getName()),
```

Comment:
You can get connector name by looking it up in output of `Metadata#getCatalogs` instead of class name of connector handle